pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -191,50 +217,71 @@ func (h *HeapsterMetricsClient) getCpuUtilizationForPods(namespace string, selec
 	}
 
 	sum := int64(0)
+	count := int64(0)
 	for _, m := range metrics.Items {
 		if _, found := podNames[m.Name]; found {
+			if _, isUnready := unreadyPods[m.Name]; isUnready {
+				// don't count unready pods
+				continue
+			}
+
 			for _, c := range m.Containers {
 				cpu, found := c.Usage[v1.ResourceCPU]
 				if !found {
 					return 0, time.Time{}, fmt.Errorf("no cpu for container %v in pod %v/%v", c.Name, namespace, m.Name)
 				}
 				sum += cpu.MilliValue()
 			}
+
+			count++
 		} else {
 			return 0, time.Time{}, fmt.Errorf("not expected metrics for pod %v/%v", namespace, m.Name)
 		}
 	}
 
-	return sum / int64(len(metrics.Items)), metrics.Items[0].Timestamp.Time, nil
+	return sum / count, metrics.Items[0].Timestamp.Time, nil
 }
 
 // GetCustomMetric returns the average value of the given custom metric from the
 // pods picked using the namespace and selector passed as arguments.
-func (h *HeapsterMetricsClient) GetCustomMetric(customMetricName string, namespace string, selector labels.Selector) (*float64, time.Time, error) {
+func (h *HeapsterMetricsClient) GetCustomMetric(customMetricName string, namespace string, selector labels.Selector) (*float64, *UtilizationInfo, error) {
 	metricSpec := getHeapsterCustomMetricDefinition(customMetricName)
 
 	podList, err := h.client.Core().Pods(namespace).List(api.ListOptions{LabelSelector: selector})
 
 	if err != nil {
-		return nil, time.Time{}, fmt.Errorf("failed to get pod list: %v", err)
+		return nil, nil, fmt.Errorf("failed to get pod list: %v", err)
 	}
 	podNames := []string{}
```

Comment:
(because of how we want to use it in getCustomMetricForPods, basically -- it saves us converting to a list for use in `Join` and the rest of the fetch logic later).